libcaf_core/caf/json_value.test.cpp

```diff
@@ -367,3 +367,78 @@ TEST("non-empty object with nested values") {
   check_eq(obj.value("min-uint64").to_unsigned(), 0u);
   check_eq(obj.value("max-uint64").to_unsigned(), UINT64_MAX);
 }
+
+std::string_view json_with_nan_or_null_values = R"_({
+  "null-value": null,
+  "nan-value": nan
+})_";
+
+TEST("non-empty object with null or nan values") {
+  auto val = unbox(json_value::parse(json_with_nan_or_null_values));
```

Comment:
This doesn't need to be global:

```suggestion
TEST("non-empty object with null or nan values") {
  std::string_view str = R"_({
    "null-value": null,
    "nan-value": nan
  })_";
  auto val = unbox(json_value::parse(str));
```

Same for the other constants.